File: judges/spoj_brasil/done/ENERGIA.cpp
#include<cstdio>
#include<vector>
using namespace std;

#define MAX 105

bool mark[MAX];
int n, m;
vector<int> g[MAX];

void dfs(int v)
{
	mark[v] = 1;
	for(int i=0; i<(int)g[v].size(); ++i)
	{
		int u = g[v][i];
		if(!mark[u]) dfs(u);
	}
}

int main()
{
	for(int tc=1; scanf("%d %d", &n, &m) && n; ++tc)
	{
		for(int i=0; i<n; ++i)
		{
			mark[i] = 0;
			g[i].clear();
		}

		while(m--)
		{
			int u, v;
			scanf("%d %d", &u, &v);
			--u; --v;
			g[u].push_back(v);
			g[v].push_back(u);
		}
		
		dfs(0);
		
		bool fail = 0;
		for(int i=0; i<n; ++i)
			if(mark[i] == 0)
			{
				fail = 1;
				break;
			}
		printf("Teste %d\n", tc);
		if(fail) puts("falha\n");
		else puts("normal\n");
	}

	return 0;
}

File: judges/spoj_brasil/done/ENERGIA.py
def dfs(v):
	mark[v] = True
	for u in g[v]:
		if mark[u] == False:
			dfs(u)

tc = 1
while True:
	n, m = map(int, raw_input().split())
	if n == 0: break
	
	g = [[] for i in range(n)]
	for i in range(m):
		u, v = map(int, raw_input().split())
		u, v = u - 1, v - 1
		g[u].append(v)
		g[v].append(u)
	
	mark = [False]*n
	dfs(0)
	
	fail = False
	for i in range(n):
		if mark[i] == False:
			fail = True
			break
	print "Teste " + str(tc)
	if fail: print "falha\n"
	else: print "normal\n"
	
	tc = tc + 1
